src/main/webapp/user-location.js

```diff
@@ -0,0 +1,60 @@
+const apiKey = 'AIzaSyB5cba6r-suEYL-0E_nRQfXDtT4XW0WxbQ';
+const fetchJson = {
+  'considerIp': 'true',
+};
+const geoCodingUrl = 'https://maps.googleapis.com/maps/api/geocode/json?latlng=';
+const geoLocationUrl = 'https://www.googleapis.com/geolocation/v1/geolocate?key=';
+
+// Given the user's lat and lng from the geoLocation API,
+// getUserState calls to the geoCoding API to reverse geoCode
+// the coordinates and get the user's current state.
+async function getUserState(lat, lng) {
+  const geoCodingFetch = geoCodingUrl + lat + ',' + lng + '&key=' + apiKey;
+  return fetch(geoCodingFetch).then(function(response) {
+      return response.json();
+    })
+    .then(function(jsonResponse) {
+      // Traverse all of the addresses and find one that
```

Comment:
We should do a check somewhere in here that the user is actually in the US, and if they are not, just return the default as being 'each U.S. state'. If I run this locally, it comes up with "Ontario" filled in - it's pretty cool, actually, but we probably don't want that.

Reply:
I was wondering what yours would return! And yes true we do not want that. Will add error checking for state and country.

Reply:
I was pretty surprised to see it there, I thought maybe there'd be an error. It was cool!

Reply:
Fixed! (hopefully)